src/logic/safe/store/reducer/moduleTransactions.ts

```diff
@@ -0,0 +1,32 @@
+import { handleActions } from 'redux-actions'
+
+import {
+  ADD_MODULE_TRANSACTIONS,
+  AddModuleTransactionsAction,
+} from 'src/logic/safe/store/actions/addModuleTransactions'
+import { ModuleTxServiceModel } from 'src/logic/safe/store/actions/transactions/fetchTransactions/loadModuleTransactions'
+
+export const MODULE_TRANSACTIONS_REDUCER_ID = 'moduleTransactions'
+
+export interface ModuleTransactionsState {
+  [safeAddress: string]: ModuleTxServiceModel[]
+}
+
+export default handleActions(
+  {
+    [ADD_MODULE_TRANSACTIONS]: (state: ModuleTransactionsState, action: AddModuleTransactionsAction) => {
+      const { modules, safeAddress } = action.payload
+      const oldModuleTxs = state[safeAddress] ?? []
+      const oldModuleTxsHashes = oldModuleTxs.map(({ transactionHash }) => transactionHash)
+      // filtering in this level happens, because backend is returning the whole list of txs
+      // so, to avoid re-storing all the txs, those already stored are filtered out
```

Comment:
why does it matter if we restore them?

Reply:
In that case, I would change the comment "to avoid duplicates..."

Reply:
indeed